### PR TITLE
Several instances of strncpy are off by one

### PR DESCRIPTION
**Pull request by adizaimi**
fix instances where strncpy can leave string without a terminating character
fix warning: pointer to void or function used in arithmetic

### CODE REVIEW

File: bdb/llmeta.c [hunk 1]
```diff
@@ -5264,10 +5265,10 @@ static int bdb_tbl_access_set(bdb_state_type *bdb_state, tran_type *input_trans,
         break;
     }
 
-    strncpy(tbl_access_data.tablename, tblname,
+    strncpy0(tbl_access_data.tablename, tblname,
```
Comment (nirbhayc) [inline bdb/llmeta.c]: Formatting

File: bdb/llmeta.c [hunk 2]
```diff
@@ -5264,10 +5265,10 @@ static int bdb_tbl_access_set(bdb_state_type *bdb_state, tran_type *input_trans,
...
             sizeof(tbl_access_data.tablename));
 
-    strncpy(tbl_access_data.username, username,
+    strncpy0(tbl_access_data.username, username,
```
Comment (nirbhayc) [inline bdb/llmeta.c]: Ditto

File: bdb/llmeta.c [hunk 3]
```diff
@@ -5477,10 +5478,10 @@ int bdb_tbl_op_access_get(bdb_state_type *bdb_state, tran_type *input_trans,
 
     tbl_access_data.file_type = LLMETA_TABLE_USER_OP;
     tbl_access_data.command_type = command_type;
-    strncpy(tbl_access_data.tablename, tblname,
+    strncpy0(tbl_access_data.tablename, tblname,
```
Comment (nirbhayc) [inline bdb/llmeta.c]: Check formatting here and more below.

File: cdb2api/cdb2api.c [hunk 4]
```diff
@@ -347,8 +348,7 @@ static char *ibm_getargv0(void)
 
     if (1 == (rc = getprocs(&p, sizeof(p), NULL, 0, &idx, 1)) &&
         _PID == p.pi_pid) {
-        strncpy(argv0, p.pi_comm, PATH_MAX);
-        argv0[PATH_MAX - 1] = '\0';
+        strncpy(argv0, p.pi_comm, PATH_MAX - 1);
```
Comment (nirbhayc) [inline cdb2api/cdb2api.c]: Why not strncpy0? I see you have included the header.

File: cdb2api/cdb2api.c [hunk 5]
```diff
@@ -1279,33 +1279,28 @@ static void read_comdb2db_cfg(cdb2_hndl_tp *hndl, FILE *fp,
             } else if (strcasecmp(SSL_CERT_PATH_OPT, tok) == 0) {
                 tok = strtok_r(NULL, " :,", &last);
                 if (tok) {
-                    strncpy(cdb2_sslcertpath, tok, PATH_MAX);
-                    cdb2_sslcertpath[PATH_MAX - 1] = '\0';
+                    strncpy(cdb2_sslcertpath, tok, PATH_MAX - 1);
```
Comment (nirbhayc) [inline cdb2api/cdb2api.c]: Ditto

File: cdb2api/cdb2api.h [hunk 6]
```diff
@@ -273,8 +273,8 @@ typedef enum cdb2_event_type {
     CDB2_AT_EXIT_NEXT_RECORD = 1 << 13,
 
     /* Lifecycle events */
-    CDB2_AT_OPEN = 1 << 30,
-    CDB2_AT_CLOSE = 1 << 31,
+    CDB2_AT_OPEN = 1 << 29,
+    CDB2_AT_CLOSE = 1 << 30,
```
Comment (nirbhayc) [inline cdb2api/cdb2api.h]: ?

File: db/types.c [hunk 7]
```diff
@@ -7368,7 +7368,7 @@ static int structdatetimeus2string_ISO(cdb2_client_datetimeus_t *in, char *out,
         if (!cdt.tzname[0])                                                    \
             strncpy(cdt.tzname,                                                \
                     ((struct field_conv_opts_tz *)outopts)->tzname,            \
-                    sizeof(cdt.tzname));                                       \
+                    sizeof(cdt.tzname) - 1);                                       \
```
Comment (nirbhayc) [inline db/types.c]: Use strncpy0 instead?

File: cdb2api/cdb2api.h (hunk 6 above)
Comment (adizaimi) [reply cdb2api/cdb2api.h]: 1 << 31 overflows the int (which is what enum is based on) and should not be used. 
We have space from 13 to 28 now.

Comment (nirbhayc) [reply cdb2api/cdb2api.h]: ICTHX

File: cdb2api/cdb2api.c (hunk 4 above)
Comment (adizaimi) [reply cdb2api/cdb2api.c]: getting 'undefined reference to `strncpy0' -- dont want to start including other libraries for this.

File: db/types.c (hunk 7 above)
Comment (adizaimi) [reply db/types.c]: Initially tried but was getting include conflicts. Trying again succeeded.

**Pull request merged**